trestle/core/duplicates_validator.py

```diff
@@ -0,0 +1,60 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2020 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validate by confirming no duplicate items."""
+
+import pathlib
+from typing import Any, Dict
+
+from trestle.core.base_model import OscalBaseModel
+from trestle.core.validator_helper import find_values_by_name
+from trestle.utils import fs
+
+
+class DuplicatesValidator:
```

Comment:
Would it make sense for this to inherit from an abstract base class the method interfaces for validate?

This would avoid the need for a builder per class as we would use the abstract base class to define the interface. It would also allow you to have a minimal set of typing as well the trade off would be that object factory is a completely generic factory which you use registation to build a specific instance.

Reply:
If we can define an interface for validators that would be fine.  Right now I am leaving it open ended since I don't know the range of scenarios or use cases.